systemd/systemd.go

```diff
@@ -62,21 +60,19 @@ func run(args ...string) ([]byte, error) {
 // systemctl. It's exported so it can be overridden by testing.
 var SystemctlCmd = run
 
-// jctl calls journalctl to get the JSON logs of the given services, wrapping the error if any.
-func jctl(svcs []string) ([]byte, error) {
-	cmd := []string{"journalctl", "-o", "json"}
-
-	for i := range svcs {
-		cmd = append(cmd, "-u", svcs[i])
+// jctl calls journalctl to get the JSON logs of the given services.
+func jctl(svcs []string, n string, follow bool) (io.ReadCloser, error) {
+	args := make([]string, 0, 2*len(svcs)+4)
```

Comment:
I like making numbers up, but in this case I know we'll have at most that many lines, at at least that many minus one¹, so doing it this way is not much more complex than starting with a nil slice, and can save a bunch of memory (as append doubles the size of the backing array every time it runs out). The added complexity of sizing it exactly right didn't seem worth it.

1. although looking at the maths now I might have not updated the number after the last tweak, I'll double-check.